### ISSUE TITLE
[lxqt-config] Add HiDPI configuration

### ISSUE THREAD

**Issue by tsimonq2** [lxqt/lxqt#1549]
Carrying over from #916, we need a menu for HiDPI configuration.

**Comment by tsujan** [comment]
Yes, it might be needed. The "release-blocker" label is an exaggeration though.

**Comment by agaida** [comment]
i would guess it will become far more important very soon - or it is right now. HiDPI devices are now far more common than a year ago. (And i'm selfish: i plan to buy a small monitor (42-43") soon - and it would be nice if i can read something on in without new glasses :P

**Comment by tsujan** [comment]
Most Qt apps support it and it's just a matter of setting `QT_SCALE_FACTOR`.

Sooner or later, it'll become automatic.

**Comment by agaida** [comment]
i know - but we should have the ability to set the scale factor (somewhere in the UI)

**Comment by agaida** [comment]
right - and re: blocker - should be not to hard to implement, but it should be done before the release, so i guess the blocker marker is right

**Comment by tsujan** [comment]
Related and not related to this: IMHO, the only blocker is the SMB problem you'd reported. Unfortunately, I can't have SMB and couldn't do anything by using your detailed explanations. It'll be bad if it isn't fixed before the next release.

**Comment by tsujan** [comment]
> should be not to hard to implement

Correct. Might look into it after becoming free from pcmanfm-qt.

**Comment by agaida** [comment]
hmm - would a virtual machine help?

**Comment by tsujan** [comment]
> hmm - would a virtual machine help?

Only if you guide me through it by giving me some links that explain it correctly. I couldn't set up SMB with VirtualBox by reading the "explanations" I found; perhaps they were too old.

**Comment by agaida** [comment]
will see what i can do

**Comment by agaida** [comment]
hmm - maybe a virtual machine is overkill - but it should possible to have that as an nspawn container - this would  be a better solution. (Does not mess up the system and one can start the container if needed)

**Comment by Vladimir-csp** [comment]
A lot of digging on font dpi was done in #980 

You can refer to hooks() function example in my [rerandr3](https://github.com/Vladimir-csp/rerandr3/blob/master/rerandr3) script for a robust mechanism of syncing font DPI for everything using xrandr as a source.

As for widget scaling, refer to [GUI Toolkits section of HiDPI ArchWiki page](https://wiki.archlinux.org/index.php/Hidpi#GUI_toolkits). It seems, Qt FINALLY got it right with `QT_AUTO_SCREEN_SCALE_FACTOR=1`. Fire and forget AT LAST, WOW!
As for GTK, two static mutually compensatory vars have to be set, i.e.: `GDK_SCALE=2`, `GDK_DPI_SCALE=0.5`

**Comment by tsujan** [comment]
@Vladimir-csp `QT_AUTO_SCREEN_SCALE_FACTOR` is exactly the opposite of what @agaida wants at https://github.com/lxqt/lxqt/issues/1571 and his point may be acceptable with big monitors. On the other hand, `QT_SCALE_FACTOR`, magnifies everything -- including point sized fonts -- and is good for most HDPI screens.

BTW, the value of `QT_SCALE_FACTOR` can be non-integer. However, there are parts of Qt that don't work well with a non-integer value -- problems are expected (I tried to find workarounds for some of them to no avail).

Anyhow, if a separate GUI is going to be made for these, `QT_AUTO_SCREEN_SCALE_FACTOR`, `QT_SCREEN_SCALE_FACTORS` and `QT_SCALE_FACTOR` should all be included in it with concise explanations.

Also a separate GUI is needed for `GDK_SCALE` and fonts of GTK apps, as in KDE.

We are talking and talking but a brave developer is needed to do the job ;) That being said, and apart from the special use case of @agaida's, users with HDPI screens can set `QT_SCALE_FACTOR` without problem because most Qt apps -- and all LXQt apps -- support it well (no blurry icons).

**Comment by Vladimir-csp** [comment]
> Also a separate GUI is needed for GDK_SCALE and fonts of GTK apps, as in KDE.

Why separate? Use the same approach as in lxqt-config-appearance.
"Fonts of GTK apps" is an irrelevant category, it's all fontconfig (>xrdb>xrandr).

**Comment by tsujan** [comment]
> Why separate?

Because Qt and GTK have separate settings. KDE's experience is good to follow too.

**Comment by tsujan** [comment]
> "Fonts of GTK apps" is an irrelevant category,

`~/.config/gtk-3.0/settings.ini`:
```
[Settings]
....
gtk-font-name=Sans 9
...
```

**Comment by Vladimir-csp** [comment]
Irrelevant in context of DPI.

**Comment by tsujan** [comment]
> Irrelevant in context of DPI.

That's correct. I meant the font and its size -- it needs a separate settings for GTK. DPI is shared.

**Comment by agaida** [comment]
We should really work on this - not only talk. It would be nice to have it in the next debian stable and lubuntu release - but hasn't in the 0.14 release, guess this would be worth a point release if we don't manage to roll it out in 0.14.
